subprojects/launcher/src/main/java/org/gradle/tooling/internal/provider/StartParamsValidatingActionExecuter.java

```diff
@@ -60,6 +62,16 @@ public Object execute(BuildAction action, BuildRequestContext requestContext, Bu
                 throw new IllegalArgumentException(String.format("The specified settings file '%s' is not a file.", startParameter.getSettingsFile()));
             }
         }
+        if (startParameter.getInitScripts().size() > 0) {
+            for (File initScript : startParameter.getInitScripts()) {
+                if (!initScript.isFile()) {
```

Comment:
It may be worth extracting this logic now into a method, given that we use it in three places now (`buildFile`, `settingFile`, `initScripts`).

Reply:
I agree, fixed in 5ac77234718f4a3e4b077ab04f86db9f83ef04cc

Reply:
👍